BaseClasses.py

```diff
@@ -1384,7 +1384,12 @@ def create_playthrough(self, create_paths: bool = True) -> None:
 
         # second phase, sphere 0
         removed_precollected: List[Item] = []
-        for item in (i for i in chain.from_iterable(multiworld.precollected_items.values()) if i.advancement):
+
+        # The lists of precollected items are going to be mutated by removing one item at a time to determine if each
+        # item is required to beat the game, and re-adding that item if it was required, so copies of the lists need to
+        # be made before iterating.
+        for item in (i for i in chain.from_iterable(items.copy() for items in multiworld.precollected_items.values())
+                     if i.advancement):
```

Comment:
```suggestion
        for item in chain.from_iterable(items.copy() for items in multiworld.precollected_items.values():
            if not item.advancement:
                continue
```

How do we feel about this? I feel like the old code was already kind of ugly and now the line break makes it even uglier

Reply:
Don't care enough to block over it, but you know, I can try

Reply:
I'd like to use two loops instead of the `chain.from_iterable`. For the initial PR, I just wanted to keep the changes small.

I think this makes it clearer to me that `precollected_items` is being modified since I don't have to figure that out from what `multiworld.precollected_items[item.player].remove(item)` does. `multiworld.push_precollected(item)` still hides away that it's modifying `precollected_items`, so I think a comment there can help too.
```py
        for precollected_items in multiworld.precollected_items.values():
            # The list of items is mutated by removing one item at a time to determine if each item is required to beat
            # the game, and re-adding that item if it was required, so a copy needs to be made before iterating.
            for item in precollected_items.copy():
                if not item.advancement:
                    continue
                logging.debug('Checking if %s (Player %d) is required to beat the game.', item.name, item.player)
                precollected_items.remove(item)
                multiworld.state.remove(item)
                if not multiworld.can_beat_game():
                    # Add the item back into `precollected_items` and collect it into `multiworld.state`.
                    multiworld.push_precollected(item)
                else:
                    removed_precollected.append(item)
```